### ISSUE TITLE
Defaults for IRF axes 

### ISSUE THREAD

**Issue by AtreyeeS** [gammapy/gammapy#2482]
We need a good choice of defaults for the IRF axes binning, `rad`, `migra` and `energy_axis_true`. Currently, `energy_axis_true` defaults to `e_reco` (PR #2479), which can be problematic in most cases, and the `rad` and `migra` axes have a fixed default (hard-coded quite randomly). These default values should ideally be chosen from the IRFs.

This is a reminder issue to discuss and implement this properly.

**Comment by cdeil** [comment]
Discussed in the Gammapy weekly call just now.

@AtreyeeS - Please create a brief overview of defaults we have now.
It could be a brief section after this one:
https://docs.gammapy.org/0.14/scripts/index.html#data-reduction-and-datasets

Probably we have duplicated and inconsistent defaults in different places, and also the choices of the defaults can be improved.

Once we have the overview, we can continue the discussion here concerning how to improve.

**Comment by AtreyeeS** [comment]
I think it would be better to edit the docs section once we have a final decision.
For now, I just summarise the present status here.

1.For 1D spectral analysis
(i) In `SpectrumDatasetMaker` : `e_true` defaults to `e_reco` unless explicitly specified.
(ii) In `SpectrumEvaluator`: `e_true` is taken from the IRFs, specifically the effective area (**even when** `e_true` is explicitly passed, the `e_true` is taken from the IRF. When no IRFs are given, the user defined value is taken.)  `e_reco` is taken from the `edisp`. If no `edisp`, then it defaults to `e_true`.
(iii) `SpectrumDataset`: Makes **no** distinction between `e_true` and `e_reco`, and takes the `energy_axis` from the `counts` or the IRFs. The `SpectrumDataset.create()` defaults `e_true` to `e_reco`, unless passed by the user.

Accessing through the high level interface, the `template-1d.yaml` does not provide any distinction between `e_true` and `e_reco`, and allows to user to specify only an `energy` axis. Thus, for practical purposes, `e_true` is always the same as `e_reco`.

2. For 3D Map  analysis
(i) The `energy_axis_true` defaults to the reco axis unless specified (in `MapDatasetMaker` and `MapDataset`.) 
(ii) Binnings for `rad` and `migra` are set completely arbitrarily to 
```
RAD_AXIS_DEFAULT = MapAxis.from_bounds(
    0, RAD_MAX, nbin=66, node_type="edges", name="theta", unit="deg"
)
MIGRA_AXIS_DEFAULT = MapAxis.from_bounds(
    0.2, 5, nbin=48, node_type="edges", name="migra"
)
```
Default for the IRF spatial binning is : 
```
BINSZ_IRF_DEFAULT = 0.2
MARGIN_IRF_DEFAULT = 0.5
```

(iii) In `simulate_dataset()`, there is no option pass the true energy axis, and all IRF maps are made on the same spatial and spectral bins as the counts `geom`.

Accessing through the high level interface,  the `template-3d.yaml` allows to explicitly specifies an `energy_axis_true`. It also allows to specify the spatial binsize (`binsz-irf`) and extent (`margin-irf`) of the IRF maps. The binnings of the `rad` and `migra` axis cannot be specified.


** Proposal **

Since this seems to be a bit messy (specially for the spectral part, we are very likely to get wrong results), I propose:

1. Unify naming between `e_true` and `energy_axis_true`....
2. Allow `e_true` to be specified in `template-1d.yaml` similar to the 3d case.
3. Adapt `simulate_dataset()` to be more flexible allowing different binning as the present `MapDataset` can accommodate.
4 (a) either hard-code a good choice of defaults for `RAD_AXIS_DEFAULT` and `MIGRA_AXIS_DEFAULT` and also (`ENERGY_AXIS_TRUE_DEFALUT`). But I don't see how we can one choice should be a  good defaults for different instruments (eg: HESS / LST /SST/ etc). So, I don't prefer this.
OR
(b) Have a helper function `get_irf_axis_defaults()` just after the datastore which returns a good choice of DEFAULTS based on the IRFs. These values are then passed to all later stages of the preparation (can be demonstrated through notebooks). The issue with this is it will probably take the IRF from the first observation, and assume that all IRFs have the same binning...
OR
(c) Assume all data reduction will ALWAYS proceed through the high level analysis unless the user really knows what he is going (and then we leave it to him to choose good binnings). Then all the axes and bin sizes can be passed through the high level interface. We suggest some "good" values, maybe by having a separate yaml file for each configuration (HESS/ LST/ etc). So add `migra-axis` and `rad-axis` as options.

Please opine @cdeil @registerrier @adonath @luca-giunti

**Comment by adonath** [comment]
During the energy dispersion check for the event sampling we noticed the migra axis default we currently have  is undersampling the energy resolution of CTA, but probably HESS as well, so we should change the defaults here. 

In general we should probably add a dictionary `IRF_AXES_DEFAULTS`, where we define the different defaults per instrument (at least "hess", "cta" and "fermi"). Proposals for the actual values are welcome anytime.

**Comment by adonath** [comment]
I think at some point we discussed introducing instrument specific default values in Gammapy for the IRFs. I still think this is a good idea. For now I would propose to keep it really simple and add a single `DEFAULT_GEOMS` somewhere in `gammapy.datasets` or possibly `gammapy.maps`.

I think it should look something like this:
```python3

DEFAULT_AXES = {
    "hess": {
        "energy_axis_true": MapAxis.from_energy_bounds(),
        "theta_axis": MapAxis.from_bounds(),
        "migra_axis": MapAxis.from_bounds(),
        "binsz_irf": "0.2 deg"
    },
    "cta": {
        "energy_axis_true": MapAxis.from_energy_bounds(),
        "theta_axis": MapAxis.from_bounds(),
        "migra_axis": MapAxis.from_bounds(),
        "binsz_irf": "0.2 deg"
    },
    "hawc": {
        "energy_axis_true": MapAxis.from_energy_bounds(),
        "theta_axis": MapAxis.from_bounds(),
        "migra_axis": MapAxis.from_bounds(),
        "binsz_irf": "1 deg"
    },
}

```

The structure is such that the dicts can be just passed on to `MapDataset.create()` in addition to the reference `geom` object.

**Comment by adonath** [comment]
@bkhelifi, @AtreyeeS, @registerrier please comment!

**Comment by registerrier** [comment]
Thanks @adonath . 
I think the proposal is good as is. Technically, I am not sure there will be anything else but `from_bounds` whatever the instrument. The IRFs given by the instrument should be the best option.
Note that `binsz_irf` is also implicitly deduced from the `offset` distribution. 

My main comment is that we then need a mechanism ensuring re-evaluating over finer true energy binning in case the number of bins per decade is insufficient.

**Comment by adonath** [comment]
@registerrier Yes, independently of this issue we can easily introduce such as mechanism, I'm just not sure whether we really need it. I think we already integrate spectral model using a log-log approach or analytically. And we probably just need a numerical integration scheme for energy dependent spatial models as well. This we could implement using e.g. `_trapz_log_log()` or based on interpolation of the cumulative distribution along the energy axis...

**Comment by adonath** [comment]
An alternative pattern could be to extract the info from the DL3 IRFs.
```python3
irf_spec = Observation.default_dataset_irf_spec
dataset = MapDataset.create(geom, **irf_spec)

geoms = Observation.default_dataset_geoms
dataset = MapDataset.from_geoms(geoms=geoms)
```

**Comment by adonath** [comment]
While this is still an important topic, it's not a blocker for v0.19. We adapt the milestone.

**Comment by adonath** [comment]
Default true energy axis should be given with `node_type="center"`, however as long as the energy binning is fine enough it does not really matter (see #1967).

**Comment by AtreyeeS** [comment]
The best defaults can be suggested by individual experiments. closing now
